crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -1628,6 +1628,42 @@ impl<'this, TX: DbTx<'this>> BlockProvider for DatabaseProvider<'this, TX> {
     fn block_body_indices(&self, num: u64) -> Result<Option<StoredBlockBodyIndices>> {
         Ok(self.tx.get::<tables::BlockBodyIndices>(num)?)
     }
+
+    fn block_with_senders(&self, block_number: BlockNumber) -> Result<Option<BlockWithSenders>> {
```

Comment:
we need to note here that the tx's do not have valid hashes somehow

Reply:
added some docs to the implementation on `DatabaseProvider`